### PR TITLE
Update: Add 'method' option to no-empty (fixes #4605)

### LINKED ISSUE #4605: Add  "No empty method" to  no-empty for ES6 methods
It would check for methods like this

```
class Person

...

  getName () {
  }
```

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/28840663-add-no-empty-method-to-no-empty-for-es6-methods?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F282608&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by kaicataldo**

### CODE REVIEW

File: docs/rules/no-empty.md [hunk 1]
```diff
@@ -65,6 +65,58 @@ try {
...
+    bar: () => {}         /*error Empty block statement.*/
+}
+```
+
+The following patterns are not considered problems when `methods` is set to `true`:
+
+```js
+/*eslint no-empty: [2, { methods: true }]*/
+
+var foo = {
```
Comment (nzakas) [inline docs/rules/no-empty.md]: The term "methods" in ES6 only refers to the shorthand notation. I don't think we should flag this or the arrow function example with this option.

Comment (kaicataldo) [inline docs/rules/no-empty.md]: Okay - Awesome, didn't know that. I'll remove the other two cases and update the docs. Should the option method be something like `es6-methods`? Or do you feel that `methods` and the proper documentation is adequate?

Comment (kaicataldo) [inline docs/rules/no-empty.md]: Updated without changing any of the language regarding specifying ES6 methods in particular - can add that in pending the response to my question above.

Comment (nzakas) [inline docs/rules/no-empty.md]: I think `methods` is fine. People will get used to this terminology eventually and the docs are there for additional guidance.

Comment (kaicataldo) [inline docs/rules/no-empty.md]: Sounds good

**Comment by nzakas** [comment]
Lgtm. Thanks!

**Pull request merged**